src/main/java/com/_up/megastore/services/implementations/ProductService.java

```diff
@@ -49,6 +49,13 @@ public Product findProductByIdOrThrowException(UUID productId) {
                 .orElseThrow(() -> new NoSuchElementException("Product with id " + productId + " does not exist."));
     }
 
+    @Override
+    public void deleteProductById(UUID productId){
+        Product product = findProductByIdOrThrowException(productId);
+        product.setDeleted(true);
+        productRepository.save(product);
```

Comment:
No está mal, pero agregaría un método más para validar que los productos que se deseen eliminar no estén ya eliminados. Esto permitiría hacer el código más legible y evitar procesamiento innecesario

```suggestion
        ifProductIsNotDeletedThrowException(productId);
        Product product = findProductByIdOrThrowException(productId);
        product.setDeleted(true);
        productRepository.save(product);
```

A esto lo podés hacer con Query Methods: https://www.baeldung.com/spring-data-derived-queries